### DISCUSSION TITLE
Simple Sphere-Sphere intersection test

### DISCUSSION THREAD

**Discussion by Constantin-Ushkov** [bepu/bepuphysics2#227] [Q&A] [answered]
Hello,
I'm very new to bepu, trying to start with solving very simple task - check if 2 spheres collide.
I see there are classes like SphereBoxTester, SphereeCapsuleeTester, etc....
But I can't find anything that is looking like what I need - a simple test of two spheres intersection...
So could anybody pls give me some directions?

P.S.
Ideeally, something as simple as possible, something like:
`if (SphereTester.Intersect(sphere1, sphere2))
{
    // do something
}`

**Comment by RossNordby** [comment, marked as answer]
While using bepuphysics2 to do something like this is possible, it'll feel a bit like using an [industrial excavator](https://github.com/bepu/bepuphysics2/blob/master/Demos/Demos/CollisionQueryDemo.cs) for casual gardening. It's meant to do much harder things in very large numbers, so the API isn't well suited to this kind of simplicity.

Instead, consider something like:

``` cs
var radiusSum = sphereRadiusA + sphereRadiusB;
var squaredRadiusSum = radiusSum * radiusSum;
if (Vector3.DistanceSquared(spherePositionA, spherePositionB) < squaredRadiusSum) { ... }
```

**Reply by Constantin-Ushkov** [reply]
Thanks for quick reply. I see your point. Well, this makes sense for sure.
But anyway, as a suggestion - I can imagine some kind of a helper that will allow such simplle tests.
It's just feels a bit weird when you reference such a powerfull and rich framework and have to write on your own a code that does simple sphere\sphere intersection test. :)

**Reply by RossNordby** [reply]
One-off boolean tests are something I've considered and do want to do for the simplicity they offer. Doing them well, however, requires implementing scalar boolean codepaths for every shape type that won't be used anywhere else in the engine. It's a pretty heavy lift, so I'm not sure when I'll get around to it.

(Technically, you can just wrap the `CollisionBatcher` in a simple scalar-style API and completely ignore the fact that it's meant for batching and computes full contact manifolds, you just won't get amazing performance.)
